packages/commercetools/composables/src/useProduct/index.ts

```diff
@@ -29,7 +29,7 @@ const productsSearch = async (params: ProductsSearchParams): Promise<ProductsSea
   };
 };
 
-const useProduct: (cacheId: string) => UseProduct<ProductVariant, Record<string, Filter>, AgnosticSortByOption[]> =
-  useProductFactory<ProductVariant, ProductsSearchParams, Record<string, Filter>, AgnosticSortByOption[]>({ productsSearch });
+const useProduct: (cacheId: string) => UseProduct<ProductVariant, Record<string, Filter>, AgnosticSortByOption[], CustomQuery> =
```

Comment:
thI think we can just type the interface of `customQuery` more loosely 9without `apolloQueryResult` and remove it from the generic interface (so have it dictated by the core, not integration developers)

Reply:
Done.